### PR TITLE
add kafka on nfs post

### PR DESCRIPTION
**Pull request by WheresAlice**

### CODE REVIEW

File: _posts/2018-07-05-kafka-nfs.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Running Kafka on this NFS mount did in fact run pretty well for quite some time.  Kafka is pretty stable day-to-day, and so we never had to rebalance anything or delete any topics.  That is until one day when we needed to expand the cluster to add some new brokers to be better prepared for some new large topics.
+
+Adding new Kafka brokers to an existing cluster doesn't actually do very much.  It makes the brokers available for producers and consumers, but any existing topics will continue to be partitioned across the original brokers.  Our two new brokers were sitting there quietly doing very little.
+
+![Unused brokers](/images/kafka-nfs/before-reallocation.png)
+
+To start using the new brokers and reduce the load on the current brokers, lets do some reallocation with this [little script](https://gist.github.com/WheresAlice/c347e6b909d0bbc4256bb891e74c3191) we've knocked together.  And immediately we start seeing errors followed by immediate crashes due to all log directories being marked as failed:
```
Comment (PhilGrayson) [inline _posts/2018-07-05-kafka-nfs.md]: The grammar in this paragraph seems difference to the previous ones. Maybe the tense (past, present) or voice (first, third) has changed.

Or might just be me...

File: _posts/2018-07-05-kafka-nfs.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Running Kafka on this NFS mount did in fact run pretty well for quite some time.  Kafka is pretty stable day-to-day, and so we never had to rebalance anything or delete any topics.  That is until one day when we needed to expand the cluster to add some new brokers to be better prepared for some new large topics.
+
+Adding new Kafka brokers to an existing cluster doesn't actually do very much.  It makes the brokers available for producers and consumers, but any existing topics will continue to be partitioned across the original brokers.  Our two new brokers were sitting there quietly doing very little.
+
+![Unused brokers](/images/kafka-nfs/before-reallocation.png)
```
Comment (PhilGrayson) [inline _posts/2018-07-05-kafka-nfs.md]: The before-reallocation.png image looks a bit complicated. Perhaps a graph of `kafka_partition_count{cluster="logs"}` will show the before and after of rebalancing.

File: _posts/2018-07-05-kafka-nfs.md (hunk 1 above)
Comment (muggahtee) [inline _posts/2018-07-05-kafka-nfs.md]: `s/lets/let's/` 🙈

File: _posts/2018-07-05-kafka-nfs.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+layout:     post
+title:      Kafka on NFS
+date:       2018-07-05
+summary:    There is a general recommendation against running Apache Kafka on NFS storage, but nobody really gives a good explanation as to why.  In this post we look at some broker crashes we have seen happening on Kafka clusters which use NFS storage and why they were happening.
+author:     alice_kaerast
+image:      kafka-nfs/before-reallocation.png
+category:   Operations
+---
+
+There is a general recommendation against running Apache Kafka on NFS storage.  But the Jira tickets relating to this are closed resolved, so why is this still a recommendation?  Surely using NFS on a nice big NAS that our storage team provide will make our Kafka cluster for log shipping nice and easy to maintain.  The logs get queued locally if Kafka is down, performance isn't a huge concern for this cluster, and we didn't know how much this cluster would grow over time.  So using NFS exports on the NAS server that our storage team provide seemed like an easy win - cheap storage, easily expandable in the future.
```
Comment (muggahtee) [inline _posts/2018-07-05-kafka-nfs.md]: Repetition of `NAS server that our storage team provide` feels a little odd

File: _posts/2018-07-05-kafka-nfs.md (hunk 2 above)
Comment (WheresAlice) [reply _posts/2018-07-05-kafka-nfs.md]: I might show that graph as after-reallocation.png at the end of the post.  There's some gaps in the graphs during broker outages, so showing that at the end would make more sense.

File: _posts/2018-07-05-kafka-nfs.md (hunk 3 above)
Comment (WheresAlice) [reply _posts/2018-07-05-kafka-nfs.md]: Yeah, I wasn't sure if that was too much repetition.  I've rewritten this bit now.

File: _posts/2018-07-05-kafka-nfs.md (hunk 1 above)
Comment (WheresAlice) [reply _posts/2018-07-05-kafka-nfs.md]: Resolved by rewriting this passive tense as per @PhilGrayson 's comment.

Comment (WheresAlice) [reply _posts/2018-07-05-kafka-nfs.md]: Rewritten to be past tense

File: _posts/2018-07-05-kafka-nfs.md (hunk 2 above)
Comment (WheresAlice) [reply _posts/2018-07-05-kafka-nfs.md]: Graph added.

**Pull request merged**